process/transaction/baseProcess.go

```diff
@@ -164,3 +170,37 @@ func (txProc *baseTxProcessor) checkTxValues(tx *transaction.Transaction, acntSn
 
 	return nil
 }
+
+func (txProc *baseTxProcessor) checkUserNames(tx *transaction.Transaction, acntSnd, acntDst state.UserAccountHandler) error {
+	isSenderUserNameCorrect := len(tx.SndUserName) > 0 &&
+		!check.IfNil(acntSnd) && !bytes.Equal(tx.SndUserName, acntSnd.GetUserName())
+	if isSenderUserNameCorrect {
```

Comment:
If isSenderUserNameCorrect you return an error? Something doesn't look right. Same for isDstUserNameCorrect. Change to isSenderUserNameWrong, or !isSenderUserNameCorrect.

Reply:
done